test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -314,7 +315,8 @@ public void RelativeApplicationBaseAreResolved()
                 .UseStartup("Microsoft.AspNetCore.Hosting.Tests")
                 .Build();
 
-            Assert.Equal(Path.GetFullPath("bar"), host.Services.GetService<IApplicationEnvironment>().ApplicationBasePath);
+            // Assert
+            Assert.EndsWith(Path.DirectorySeparatorChar + "bar", host.Services.GetRequiredService<IApplicationEnvironment>().ApplicationBasePath);
```

Comment:
Check at least if the path is rooted otherwise this test no longer tests what it is supposed to test.